app/services/foreman/foreman_url_renderer.rb

```diff
@@ -39,20 +36,44 @@ def foreman_url(action, params = {}, unescaped_params = {})
         raw_string += unescaped_params.map { |k, v| "#{k}=#{v}" }.join('&')
       end
 
+      url_options[:action] = action
+      url_options[:path] = config.path
+      render_foreman_url(host, url_options, params) + raw_string
+    end
+
+    apipie :method, 'Returns Foreman request address' do
+      desc 'Returns FQDN + Port to Foreman or Smart Proxy depending on host.subnet.template proxy configuration.'
```

Comment:
Maybe we could rephrase it a bit, what do you think?
```suggestion
      desc 'Returns Foreman or Smart Proxy FQDN with port depending on host.subnet.template proxy configuration.'
```